src/04-modality-specific-files/11-functional-near-infrared-spectroscopy.md

```diff
@@ -0,0 +1,323 @@
+NIRSCoordinateSystemUnits fields in `*_coordsystem.json`. If an `*_optodes.tsv`
+file is specified, a `*_coordsystem.json` file MUST be specified as well.
+The order of the required columns in the `*_optodes.tsv` file MUST be as listed below.
+
+The x, y, and z positions are for measured locations, for example, with a polhemus
+digitizer. If you also have idealized positions, where you wish the optodes to be
+placed, these can be listed in the template values
+(for example for "template positions" computed on a sphere).
+SNIRF contains arrays for both
+the 3D and 2D locations of data. In BIDS the `*_optodes.tsv` file MUST contain the 3D locations. Only in case 3D positions are unavailable the 2D locations should be used, setting the z field to an `n/a` value.
+
+The following columns MUST be present:
+
+{{ MACROS___make_columns_table(
+   {
+      "name__optodes": "REQUIRED",
+      "type__optodes": "REQUIRED",
+      "x__optodes": "REQUIRED",
+      "y__optodes": "REQUIRED",
+      "z__optodes": "REQUIRED",
+   }
+) }}
+
+The following columns MAY be present:
+
+{{ MACROS___make_columns_table(
+   {
+      "x__template": "OPTIONAL but REQUIRED if `x` is `n/a`",
+      "y__template": "OPTIONAL but REQUIRED if `x` is `n/a`",
+      "z__template": "OPTIONAL but REQUIRED if `x` is `n/a`",
+      "description__optode": "OPTIONAL",
+      "detector_type": "OPTIONAL",
+      "source__optodes": "OPTIONAL",
+   }
+) }}
+
+Example:
+```Text
+name    type         x          y         z          template_x    template_y   template_z
+A1      source       -0.0707    0.0000    -0.0707    -0.07         0.00         0.07
+Fz      detector     0.0000     0.0714    0.0699     0.0           0.07         0.07
+S1      source       -0.2707    0.0200    -0.1707    -0.03         0.02         -0.2
+D2      detector     0.0022     0.1214    0.0299     0.0           0.12         0.03
+VisS2   source       -0.1707    0.1200    -0.3707    -0.1          0.1          -0.4
+VisD4   detector     0.0322     0.2214    0.2299     0.02          0.22         0.23
+```
+
+## Coordinate System JSON (`*_coordsystem.json`)
+
+{{ MACROS___make_filename_template(datatypes=["nirs"], suffixes=["coordsystem"]) }}
+
+A `*_coordsystem.json` file is used to specify the fiducials, the location of anatomical landmarks,
+and the coordinate system and units in which the position of optodes and landmarks is expressed.
+Fiducials are objects with a well-defined location used to facilitate the localization of sensors
+and co-registration, anatomical landmarks are locations on a research subject such as the nasion
+(for a detailed definition see the coordinate system description in the BIDS specification).
+The `*_coordsystem.json` is REQUIRED if the optional `*_optodes.tsv` is specified. If a corresponding
+anatomical MRI is available, the locations of anatomical landmarks in that scan should also be stored
+in the `*_T1w.json` file which goes alongside the fNIRS data.
```

Comment:
I would need an example here to understand what's being proposed, but this feels like it's going to be extraordinarily difficult to validate. What is a corresponding anatomical MRI? Any T1w image in the subject, or one linked by some metadata? What happens if there are two T1w images? What should the additional metadata be?

Reply:
As @sappelhoff notes in https://github.com/bids-standard/bids-specification/pull/802#pullrequestreview-1121994111:

> HOWEVER, if this is what's also in MEG, EEG, etc. (which I suspect), then it's fine leaving it as is (for consistency) and tackling it all together in an upcoming PR at some point (doesn't need to be done by NIRS team)

I just checked, and this same text is in EEG. It is NOT in iEEG or MEG.

@sappelhoff I am thinking of just removing this sentence, do you foresee any issues with that?

Reply:
No, please do leave it aligned with the specification for [EEG](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/03-electroencephalography.html#coordinate-system-json-_coordsystemjson), as scalp optode localization compares to EEG scalp electrode localization. For MEG (using head coils) and iEEG (using CT imaging data) the localization procedures are different. 
@effigies the corresponding T1w is the one that is pointed to with the `IntendedFor` in the `coordinates.json`

Reply:
Thanks @robertoostenveld . Marking this as resolved as IntendedFor was added today.